client/src/components/squids/SquidForm.jsx

```diff
@@ -0,0 +1,131 @@
+/* eslint-disable no-nested-ternary */
+import React, { useState } from "react";
+
+import { useMutation, queryCache } from "react-query";
+
+import { ApiClient } from "../../backend/ApiClient";
+
+import "../../style/squids/squidForm.pcss";
+
+import { useForm } from "react-hook-form";
+
+import { FormError } from "../common/FormError";
+import { usePostSquidMutation } from "../hooks/usePostSquidMutation";
+
+export const SquidForm = ({ setRefreshMessage, setFormSuccess, formSuccess }) => {
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    reset,
+  } = useForm({
+    defaultValues: {
+      experiencePoints: 0,
+    },
+  });
+  const postSquidMutation = usePostSquidMutation({ setRefreshMessage, reset });
+
+  const onSubmit = async (data) => {
+    postSquidMutation.mutate(data);
```

Comment:
You can get around having to pass callbacks into usePostSquidMutation by providing them here instead! 

```
postSquidMutation.mutate(data, { onSettled: () ...
```